docs/src/modules/concepts/pages/inter-agent-comms.adoc

```diff
@@ -0,0 +1,93 @@
+
+include::ROOT:partial$include.adoc[]
+
+Agents can collaborate in many ways, from orchestration, to direct communication, to other patterns like swarms and teams. Many of these communication patterns make for great demos, but don’t scale well or run safely and securely in production environments.
+
+Further, the emergence of https://modelcontextprotocol.io/introduction[MCP, window="new"], https://a2aprotocol.ai/[A2A, window="new"], and https://agentcommunicationprotocol.dev/introduction/welcome[ACP, window="new"] have made it (perhaps too) easy to enable direct communication between agentic services. This has led to many techniques for agent-to-agent communications: some extremely valuable and some deceptively dangerous. While Akka makes it simple to implement any of these techniques, it is worth exploring each of these patterns and anti-patterns in detail.
+
+The following table provides a quick reference as to which communication paths are preferred patterns versus ones you should avoid. Additional details are provided thereafter.
+
+[cols="1,3m,6"]
+|===
+| | Path | Description
+
+|✅| Workflow -> Agent(s) +
+Workflow -> A2A/ACP -> Agents +
+Workflow -> Endpoint (API) | Workflows are engines of orchestration that can ensure the successful outcome of many invocations across agents that are local or remote. Workflows have built-in logic for handling retries and failure for long-running processes. 
+
+|✅| Agent -> Broker +
+Workflow -> Broker +
+Entity -> Broker | Leverage an event-driven architecture that completely disconnects the awareness of one service that is producing information from those that consume it. However, for agent-to-agent communications where agents are part of a single system working towards a common goal, broker-based architectures eliminate group-based goals, shared state, and system-wide guardians. + 
+Event-driven architectures are promoted for their scale and autonomy, but create separation that makes it harder for autonomous systems to collaborate towards a common goal. 
+
+|✅| Broker -> Consumer -> Agent | Agents can be tasked to execute goals triggered by events that are sent through a broker. Provides independent intelligence and autonomy. Suffers from the same “unable to work towards a broader goal” problem.
+
+|⛔| Agent -> Agent +
+Endpoint (API) -> Agent | Avoid leveraging protocols such as HTTP, JSON-RPC, gRPC, A2A, and ACP that enable direct communication and invocation of one agent to another. 
+
+|⛔| Agent -> MCP Server -> Agent | It is best to avoid using an agent’s ability to call tools or services as a way to reach other agents. This creates indirect connections that reduce clarity and make the system harder to reason about.
+
+|⛔| Endpoint (API) -> Endpoint (API) | Avoid having an endpoint call functions directly on another endpoint. This can create side effects that aren't captured by any of the durable event flows and can bypass the resiliency and scalability provided by workflows.
```

Comment:
```suggestion
|⛔| Endpoint (API) -> Endpoint (API) | Avoid having an endpoint call functions directly on another endpoint. This can create side effects that aren't captured by any of the durable event flows and can bypass the resiliency and scalability provided by Workflows.
```